common/cpp/src/google_smart_card_common/value.h

```diff
@@ -72,10 +72,15 @@ class Value final {
   explicit Value(BinaryStorage binary_value);
   explicit Value(DictionaryStorage dictionary_value);
   explicit Value(ArrayStorage array_value);
+  // Forbid construction from pointers other than `const char*`. Without this
+  // deleted overload, the `bool`-argument version would be silently picked up.
+  explicit Value(void*) = delete;
```

Comment:
Self-review: this actually has to be `const void*` in order to catch more cases. Ditto in value_conversion.h.